Source/WTF/wtf/Threading.h

```diff
@@ -109,6 +109,11 @@ class WTF_CAPABILITY("is current") Thread : public ThreadSafeRefCounted<Thread>
     friend class ThreadGroup;
     friend WTF_EXPORT_PRIVATE void initialize();
 
+    class ApplicationData : public ThreadSafeRefCounted<ApplicationData> {
```

Comment:
I don’t think _application_ is a great term to use here. I understand we need some abstract way to refer to `WebCore` from `JavaScriptCore`, but _application_ has a pretty specific meaning as a term of art on platforms like iOS, and `WebCore` is not one of those things.

Why does this need to be reference counted? What is the second owner besides the `Thread`?

Reply:
How about ClientData?

Reply:
`ClientData` sounds good.

> Why does this need to be reference counted? What is the second owner besides the Thread?

Because WebThread is sharing it with the main thread. So, in iOS, WebKitLegacy, it is shared by multiple threads (WebThread and OS main thread).